REPO-5614 New approach for code generation

jlosornogil opened the pull request:
Before we include integration tests this PR is proposing a new approach to generate the client projects source code and build the project.

The idea is to get rid of the `generate.sh` scripts and incorporate all the logic to generate the source code and build the project in the maven lifecycle.

josorno-ccc committed:
REPO-5614 Move generate.sh scripts to maven build lifecycle

Do not ignored in git generated projects pom files to avoid issues triggering the maven
build.
Move the functionality included in the generate.sh files to the maven build lifecycle.

josorno-ccc committed:
REPO-5614 Remove generate.sh scripts

Remove unnecessary generate.sh scripts and update the travis configuration.

josorno-ccc committed:
REPO-5614 Remove generate maven profile

Just execute the client code generation in the main maven build, not
requiring the use of a maven profile.

josorno-ccc committed:
Merge branch 'develop' into dev-jlosornogil-REPO-5614-maven-compilation

josorno-ccc committed:
REPO-5614 Fix travis configuration

josorno-ccc committed:
REPO-5614 Update travis configuration

josorno-ccc committed:
REPO-5614 Update travis config

josorno-ccc committed:
REPO-5614 Update travis configuration

mmartinadanx commented on .travis.yml:
```diff
@@ -1,25 +1,6 @@
-  - source: Alfresco/alfresco-build-tools:.travis.pre-commit.yml@v1.1.2
+  - source: Alfresco/alfresco-build-tools:.travis.java.yml@v1.1.5
   - source: Alfresco/alfresco-process-scripts:slack_apa_channel_notification.yml@master
-jobs:
```
Perfect. Since maven itself is generating the code as part of the build, we don't need to override those stages any more

jlosornogil replied:
Exactly, that's the idea

mmartinadanx commented on .travis.yml:
```diff
@@ -30,5 +11,4 @@ before_install:
       -Dacs.version=${ACS_VERSION:-7.1.0.1}"
-  -  if [ "$TRAVIS_JOB_NAME" = "build application" ]; then (cd alfresco-apa-java-rest-api/alfresco-apa-java-rest-api-lib; sh generate.sh); fi
-  -  if [ "$TRAVIS_JOB_NAME" = "build application" ]; then (cd alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib; sh generate.sh); fi
+install: mvn $MAVEN_CLI_OPTS install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
```
By default, the code will be generated every time the project is built. It would be great to have an option to skip the generation. E.g. when we are developing the event client. 

`mvn clean install -Dskip.generation`

jlosornogil replied:
Yes, that's a very good idea. I'm going to bring back the `generate` maven profile, enable it by default and use the system variable `skip.generation` to disable it.

jesty reviewed (approved):
LGTM

josorno-ccc committed:
REPO-5614 Allow skipping generate step

Bring back the generate maven profile and enable it whenever the maven
property skip.generation is not set.
